src/scenes/Event/scenes/oracle.js

```diff
@@ -589,74 +539,77 @@ export default class OraclePage extends React.Component {
     if (token === Token.Qtum && status === OracleStatus.WaitResult && resultSetterQAddress !== lastUsedAddress) {
       return {
         disabled: true,
-        message: <FormattedMessage
-          id="oracle.cOracleDisabledText"
-          defaultMessage="You are not the result setter for this Event. You must wait until they set the result, or until the Open Result Set start time begins."
-        />,
+        id: 'oracle.cOracleDisabledText',
+        message: 'You are not the result setter for this Event. You must wait until they set the result, or until the Open Result Set start time begins.',
         warningTypeClass: EventWarningType.Info,
       };
     }
 
-    // Did not select a result
-    if (!(token === Token.Bot && status === OracleStatus.WaitResult) && currentOptionIdx === -1) {
-      return {
-        disabled: true,
-        message: <FormattedMessage
-          id="oracle.selectResultDisabledText"
-          defaultMessage="Please click and select one of the options."
-        />,
-        warningTypeClass: EventWarningType.Info,
-      };
-    }
-
-    // Did not enter an amount
-    if (status === OracleStatus.Voting && (voteAmount <= 0 || Number.isNaN(voteAmount))) {
+    // Trying to set result or vote when not enough QTUM or BOT
+    const totalQtum = _.sumBy(walletAddresses, (wallet) => wallet.qtum ? wallet.qtum : 0);
+    const filteredAddress = _.filter(walletAddresses, { address: lastUsedAddress });
+    const currentBot = filteredAddress.length > 0 ? filteredAddress[0].bot : 0; // # of BOT at currently selected address
+    if ((token === Token.Bot || (token === Token.Qtum && status === OracleStatus.OpenResultSet))
+      && (totalQtum < maxTransactionFee && currentBot < oracle.consensusThreshold)
+      && [OracleStatus.Voting, OracleStatus.WaitResult, OracleStatus.OpenResultSet].includes(status)) {
       return {
         disabled: true,
-        message: <FormattedMessage
-          id="oracle.enterAmountDisabledText"
-          defaultMessage="Please entered a valid amount."
-        />,
-        warningTypeClass: EventWarningType.Info,
+        id: 'str.notEnoughQtumAndBot',
+        message: 'You don\'t have enough QTUM or BOT',
+        warningTypeClass: EventWarningType.Error,
       };
     }
 
     // Trying to bet more qtum than you have
-    const totalQtum = _.sumBy(walletAddresses, (wallet) => wallet.qtum ? wallet.qtum : 0);
-    if (token === Token.Qtum && status === OracleStatus.Voting && voteAmount > totalQtum) {
+    if ((status === OracleStatus.Voting && voteAmount > totalQtum + maxTransactionFee)
```

Comment:
you need to check the token type.
`token === Token.Qtum && status === OracleStatus.Voting` = Bet phase
`token === Token.Bot && status === OracleStatus.Voting` = Vote phase

see above comment about rules